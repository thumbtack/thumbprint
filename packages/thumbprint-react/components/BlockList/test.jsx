import React from 'react';
import { mount } from 'enzyme';
import { BlockList, BlockListItem, BlockListItemLink } from './index';

describe('BlockList', () => {
    test('renders content that is passed in', () => {
        const wrapper = mount(<BlockList>goose</BlockList>);
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });
    test('renders a bottom border on the list', () => {
        const wrapper = mount(<BlockList border="bottom">goose</BlockList>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders a border around list', () => {
        const wrapper = mount(<BlockList border="group">goose</BlockList>);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('BlockListItem', () => {
    test('renders content that is passed in', () => {
        const wrapper = mount(<BlockListItem>goose</BlockListItem>);
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });
});

describe('BlockListItemLink', () => {
    test('renders content that is passed in', () => {
        const wrapper = mount(
            <BlockListItemLink href="https://thumbtack.com">goose</BlockListItemLink>,
        );
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });
    test('renders a class that is passed in', () => {
        const wrapper = mount(
            <BlockListItemLink href="https://thumbtack.com" className="red">
                goose
            </BlockListItemLink>,
        );
        expect(wrapper.find('a').hasClass('red')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders a link', () => {
        const wrapper = mount(
            <BlockListItemLink href="https://thumbtack.com">goose</BlockListItemLink>,
        );
        expect(wrapper.find('a').prop('href')).toBe('https://thumbtack.com');
        expect(wrapper).toMatchSnapshot();
    });
    test('forwards custom props', () => {
        const wrapper = mount(
            <BlockListItemLink href="https://thumbtack.com" target="_blank">
                goose
            </BlockListItemLink>,
        );
        expect(wrapper.find('a').prop('target')).toBe('_blank');
        expect(wrapper).toMatchSnapshot();
    });
});
