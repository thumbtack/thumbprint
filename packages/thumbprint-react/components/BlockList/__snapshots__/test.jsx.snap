// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockList renders a border around list 1`] = `
<BlockList
  border="group"
>
  <div
    className="list listBorderGroup"
  >
    goose
  </div>
</BlockList>
`;

exports[`BlockList renders a bottom border on the list 1`] = `
<BlockList
  border="bottom"
>
  <div
    className="list listBorderBottom"
  >
    goose
  </div>
</BlockList>
`;

exports[`BlockList renders content that is passed in 1`] = `
<BlockList>
  <div
    className="list"
  >
    goose
  </div>
</BlockList>
`;

exports[`BlockListItem renders content that is passed in 1`] = `
<BlockListItem>
  <div
    className="item"
  >
    goose
  </div>
</BlockListItem>
`;

exports[`BlockListItemLink forwards custom props 1`] = `
<BlockListItemLink
  href="https://thumbtack.com"
  target="_blank"
>
  <a
    className="itemLink"
    href="https://thumbtack.com"
    target="_blank"
  >
    goose
  </a>
</BlockListItemLink>
`;

exports[`BlockListItemLink renders a class that is passed in 1`] = `
<BlockListItemLink
  className="red"
  href="https://thumbtack.com"
>
  <a
    className="itemLink red"
    href="https://thumbtack.com"
  >
    goose
  </a>
</BlockListItemLink>
`;

exports[`BlockListItemLink renders a link 1`] = `
<BlockListItemLink
  href="https://thumbtack.com"
>
  <a
    className="itemLink"
    href="https://thumbtack.com"
  >
    goose
  </a>
</BlockListItemLink>
`;

exports[`BlockListItemLink renders content that is passed in 1`] = `
<BlockListItemLink
  href="https://thumbtack.com"
>
  <a
    className="itemLink"
    href="https://thumbtack.com"
  >
    goose
  </a>
</BlockListItemLink>
`;
